OrbitCore/BpfTrace.cpp

```diff
@@ -262,3 +286,158 @@ void BpfTrace::CommandCallbackWithCallstacks(const std::string& a_Line)
         return;
     }
 }
+
+
+//-----------------------------------------------------------------------------
+void BpfTrace::RunPerfEventOpen(bool* a_ExitRequested)
+{
+    #if __linux__
+    int ROUND_ROBIN_BATCH_SIZE = 5;
+    std::vector<uint64_t> fds;
+    std::map<Function*, LinuxPerfRingBuffer> uprobe_ring_buffers;
+    std::map<Function*, LinuxPerfRingBuffer> uretprobe_ring_buffers;
+    
+    for (const auto& pair : Capture::GSelectedFunctionsMap)
+    {
+        // gather function information
+        Function* function = pair.second;
+        uint64_t offset = function->m_Address;
+        uint64_t virtual_address = function->GetVirtualAddress();
+        std::string module = ws2s(function->m_Pdb->GetFileName());
+
+        // create uprobe for that function on that PID profiling all cpus
+        uint64_t uprobe_fd = 0;
+        if (GParams.m_BpftraceCallstacks)
+            uprobe_fd = LinuxPerfUtils::uprobe_event_open(
+                module.c_str(), offset, Capture::GTargetProcess->GetID(), -1, 
+                PERF_SAMPLE_STACK_USER | PERF_SAMPLE_REGS_USER
+            );
+        else
+            uprobe_fd = LinuxPerfUtils::uprobe_event_open(
+                module.c_str(), offset, Capture::GTargetProcess->GetID(), -1
+            );
+
+        uprobe_ring_buffers.emplace(function, uprobe_fd);
+        fds.push_back(uprobe_fd);
+
+        // create uretprobe for that function on that PID profiling all cpus
+        uint64_t uretprobe_fd = LinuxPerfUtils::uretprobe_event_open(
+            module.c_str(), offset, Capture::GTargetProcess->GetID(), -1
+        );
+        uretprobe_ring_buffers.emplace(function, uretprobe_fd);
+        fds.push_back(uretprobe_fd);
+
+        // start capturing
+        LinuxPerfUtils::start_capturing(uprobe_fd);
+        LinuxPerfUtils::start_capturing(uretprobe_fd);
+    }
+
+    LinuxPerfEventProcessor event_buffer(std::make_unique<BpfTraceVisitor>());
+
+    bool new_events = false;
```

Comment:
new_events is always false.

Reply:
See [comment](https://github.com/pierricgimmig/orbitprofiler/pull/89#discussion_r354154537) above.